src/osd/OSDCap.cc

```diff
@@ -336,6 +336,8 @@ void OSDCapGrant::expand_profile()
                                 OSDCapSpec(osd_rwxa_t(OSD_CAP_CLS_R)));
     profile_grants.emplace_back(OSDCapMatch(string(), "rbd_mirroring"),
                                 OSDCapSpec(osd_rwxa_t(OSD_CAP_CLS_R)));
+    profile_grants.emplace_back(OSDCapMatch(string(), "rbd_info"),
+                                OSDCapSpec(osd_rwxa_t(OSD_CAP_CLS_R)));
```

Comment:
We actually just want to restrict to invoking "rbd.metadata_list" cls method (see https://github.com/ceph/ceph/pull/28253/files#diff-1ad4853f970880c78ea0e52c81e621b4L351 below for an example of how to restrict by class method)

Reply:
... and the ```OSDCapMatch(string(), ...``` should copy the pool restriction (if any) from the profile cap so it it doesn't unnecessarily give global access to that object.

Reply:
> ... and the `OSDCapMatch(string(), ...` should copy the pool restriction (if any) from the profile cap so it it doesn't unnecessarily give global access to that object.

@dillaman  Updated, I am not sure if I got your point of "copy the pool restriction (if any) from the profile cap". Thank you.

Reply:
See [1] where the pool/namespace restriction is applied to the ```OSDCapMatch```. The difference is that we would only want to include the pool restriction (not the ns restriction), so you could pass ```OSDCapMatch(profile.pool_namespace.pool_name)```

[1] https://github.com/ceph/ceph/pull/28253/files#diff-1ad4853f970880c78ea0e52c81e621b4R348